sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -381,17 +382,32 @@ public void testUpdateNestedObjectDeleteWithoutArgs() throws Exception {
     }
 
     @Test
-    public void testUpdateWithNestedObjectArrayIdxAccess() throws Exception {
+    public void testUpdateArrayElementIsNotAllowed() {
         execute("create table test (coolness array(float)) with (number_of_replicas=0)");
         ensureYellow();
         execute("insert into test values (?)", new Object[]{new Object[]{2.2, 2.3, 2.4}});
-        assertEquals(1, response.rowCount());
         refresh();
 
         expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("Array index must be in range 1 to 2147483648");
+        expectedException.expectMessage("Updating a single element of array is not supported");
+        execute("update test set coolness[1] = 3.3");
+    }
+
+    @Test
+    public void testUpdateArrayUsingArrayCatFunction() {
```

Comment:
hm as col = array_cat() is just processed like any expression assignment, I'm not sure if we need a dedicated test for it